contracts/strategies/allocation/AllocationWithVotingBase.sol

```diff
@@ -0,0 +1,306 @@
+        applicationStatus[applicationId] = IAllocationStrategy.ApplicationStatus.Pending;
+
+        emit ApplicationSubmitted(applicationId, identityId, recipient, metadata, _sender);
+
+        return applicationId;
+    }
+
+    /// @notice Returns the status of the application
+    /// @param applicationId The applicationId of the application
+    function getApplicationStatus(uint256 applicationId) external view override returns (ApplicationStatus) {
+        return applicationStatus[applicationId];
+    }
+
+    /// @notice Allocates votes to the application(s)
+    /// @param _data The data to be decoded
+    /// @param sender The sender of the transaction
+    function allocate(bytes memory _data, address sender) external payable override onlyAllo returns (uint256) {
+        if (votingStartTime > block.timestamp || votingEndTime < block.timestamp) {
+            revert VOTING_NOT_OPEN();
+        }
+
+        // Decode the _data
+        // Note: should have option to vote on multiple applications so an array of Payouts is expected.
+        (Transfer.Payout[] memory payouts, address token, uint256[] memory applicationIds) =
+            abi.decode(_data, (Transfer.Payout[], address, uint256[]));
+
+        if (payouts.length > 1) {
+            // Transfer funds
+            // Note: this should live in distribution strategy
+            // bool transferSuccess = Transfer.transferPayouts(token, payouts);
+            // if (!transferSuccess) {
+            //     revert Transfer.TRANSFER_FAILED();
+            // }
+
+            // Loop through the payouts and update the votes on each application
+            for (uint256 i; i < payouts.length; i++) {
+                for (uint256 a; a < applicationIds.length; a++) {
+                    if (payouts[i].to == applications[applicationIds[a]].recipient) {
+                        Application storage application;
+                        application = applications[applicationIds[a]];
+                        application.votes = payouts[i].amount;
+                        application.amount += payouts[i].amount;
+                        emit ApplicationVoted(
+                            sender,
+                            applications[applicationIds[a]].applicationId,
+                            payouts[i].amount,
+                            token,
+                            application.recipient
+                        );
+                    }
+                    // No matching application found
+                }
+            }
+
+            return 1;
+        } else {
+            // Transfer funds
+            // Note: this should live in distribution strategy
+            // bool transferSuccess = Transfer.transferPayout(token, payouts[0]);
+            // if (!transferSuccess) {
+            //     revert Transfer.TRANSFER_FAILED();
+            // }
+
+            Application storage application;
+            application = applications[applicationIds[0]];
```

Comment:
redundant code block

Reply:
Not sure if this is still the case, some updates have been made since and don't see it.

Reply:
it's already fixed @codenamejason